docs/developer-env.md

```diff
@@ -0,0 +1,534 @@
+
+1. Open the App Store on your Mac.
+1. Search for Xcode.
+1. Select **Install**.
+1. Once complete, open a new terminal window and run `git version` to verify Git was installed.
+
+```
+
+```{group-tab} Linux
+
+Git is already installed on most Linux Debian/Ubuntu distributions.
+
+To install the latest stable Git version on Linux Debian/Ubuntu distributions:
+
+1. Open a terminal window and run `sudo apt-get install git-all`.
+1. Once complete, run `git version` to verify Git was installed.
+
+See [git-scm documentation](https://git-scm.com/downloads/linux) for more information about installing Git on other Linux distributions.
+
+```
+
+````
+
+(wsl)=
+
+## Windows Subsystem for Linux
+
+Developers can access the power of both Windows and Linux on a Windows machine. The Windows Subsystem for Linux (WSL) lets developers install a Linux distribution and use Linux applications, utilities, and Bash command-line tools directly on Windows without the overhead of a virtual machine or dual-boot setup.
+
+WSL is an optional feature on Windows 10 version 2004 and higher (Build 19041 and higher) or Windows 11. You can enable it through PowerShell or Windows Command Prompt. The steps below outline the recommended setup.
+
+<!---
+### Windows Features dialog
+
+To enable WSL on Windows using Windows Features dialog:
+
+1. In the Windows search bar, enter 'features' to bring up the **Turn Windows Features on and off** dialog.
+1. Scroll down and check **Windows Subsystem for Linux**.
+1. Select **OK**.
+1. After restarting Windows, open a PowerShell or Command Prompt and type `wsl` to confirm that you have WSL enabled.
+
+To install Ubuntu on WSL:
+
+1. Go to **Start Button > Microsoft Store**.
+1. Enter 'Linux' into the search field, then click **Run Linux on Windows**.
+1. Select the latest Ubuntu distribution.
+1. Select **Get** and wait for Windows to download and install Ubuntu.
+1. When it’s finished, select **Launch**.
+1. A terminal window will appear. Wait for Ubuntu to finish installing, then create a new Linux username and password when prompted.
+
+### PowerShell or Windows Command Prompt
+--->
+
+To enable WSL on Windows using Powershell or Windows Command Prompt:
+
+1. Right-click and select **Run as administrator** to use PowerShell or Windows Command Prompt in administrator mode.
+1. Run `wsl --install`.
+
+    :::{note}
+    This command will enable the features necessary to run WSL and install the Ubuntu distribution.
+    :::
+
+1. When prompted, restart Windows.
+1. After restarting Windows, open the Ubuntu distribution and create a new Linux **User Name** and **Password** when prompted.
+
+    :::{note}
+    The **User Name** and **Password** is specific to each Linux distribution that you install and has no bearing on your Windows user name.
+    :::
+
+See [Set up a WSL development environment](https://learn.microsoft.com/en-us/windows/wsl/setup/environment) for more information.
+
+## Development Containers
```

Comment:
I wonder if this section would be better off as a blog post that we can point to in an admonition from this page.

(a) This part feels quite a lot more detailed than the rest of the page and (b) I feel that devcontainers are more of a "this can be a nice thing" rather than a requirement. I, and I guess the majority, of Nextflow developers have never used this.

Reply:
I think it's emerging but not especially common. I'll drop it from this page and add it to the list of possible blog posts (maybe something @kenibrewer would want to write)

Reply:
We also have a "Tutorials" section in the Nextflow docs as a temporary home for tutorial-like content until we have something better on seqera.io, so if you want something more permanent than a blog post you can put it there.

Blog post or tutorial would be more appropriate for dev containers for now IMO

Reply:
Okay, great. I think it's agreed that it should be removed from the PR. I'll keep the text in case we can recycle it for a blog or tutorial. Depending on how it's pitched, it could be wrapped up into a tutorial or blog. Either way, we can put it aside for now.

Reply:
I've commented this out for now. I'll keep the text incase someone wants to write a blog in the new year.